core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,369 @@
+    # File needs the MAF file extension, otherwise it is not recognized by SigProfilerMatrixGenerator
+    outfile = tmp_dir + "/mutations.maf"
+    maf.to_csv(outfile, sep="\t", index=False)
+    return outfile, genome
+
+
+def genome_install(genome):
+    """Install a genome for SigProfilerMatrixGenerator
+    
+    :param genome: genome build (GRCh37 or GRCh38)
+    :return: None    
+    """
+    genInstall.install(genome)
+    return None
+
+
+def run_matrix_generator(genome_build, maf_folder):
+    """Run SigProfilerMatrixGenerator to generate substitution matrices in input folder
+    
+    :param genome_build: str, genome build (GRCh37 or GRCh38)
+    :param maf_folder: str, path to folder containing maf input files
+    :return: None
+    """
+    print("-" * 60 + "\nGenerating nucleotide matrices: ")
+    # Name of the project is irrelevant, these files will be removed after MS algorithm has been run
+    project = "tmp"
+    matGen.SigProfilerMatrixGeneratorFunc(project, genome_build, maf_folder)
+    return None
+
+
+def install_temposig():
+    """Call R to install tempoSig
+    
+    :return: None
+    """
+    robjects.r('devtools::install_github("mskcc/tempoSig")')
+    return None
+
+
+def check_install_temposig():
+    """Call R to check if tempoSig is installed and where, otherwise install using R
+    
+    :return: str, location of tempoSig.R script
+    """
+    temposig_location = robjects.r("system.file('exec', 'tempoSig.R', package = 'tempoSig')")[0]
+    if not temposig_location:
+        install_temposig()
+        temposig_location = robjects.r("system.file('exec', 'tempoSig.R', package = 'tempoSig')")[0]
+        if not temposig_location:
+            raise ImportError("Unable to install or locate the tempoSig R-package. Please install manually. "
+                              "If this error persists, either use the --temposig parameter or use another algorithm.")
+    return temposig_location
+
+
+def matrix_temposig(file_name):
+    """Remove the first column on the first row, tempoSig requires this
+
+    :param file_name: str, name of matrix file to alter
+    :return: None, file is altered on host machine
+    """
+    with open(file_name, "r") as infile:
+        data = infile.readlines()
+    data[0] = "\t".join(data[0].split("\t")[1:])
+    with open(file_name, "w") as outfile:
+        outfile.writelines(data)
+    return None
+
+
+def run_temposig(temposig_loc, inf_matrix, outf, cosmic=2, nperm=1000):
+    """Generate mutational signatures using tempoSig
+    
+    :param temposig_loc: str, path to tempoSig.R script
+    :param inf_matrix: str, path to input mutational matrix
+    :param outf: str, path to output file
+    :param cosmic: int, cosmic version to use (2, 3)
+    :param nperm: int, number of permutations for p-value estimation (default: 1000)
+    :return: None
+    """
+    command = f"{temposig_loc} --cosmic_v{cosmic} --pvalue --nperm {nperm} --cbio {inf_matrix} " \
```

Comment:
What do you mean with making temposig external? Not performing the installation? Or also letting the user run temposig themselves, instead of from the script?